app/ui/browser/actions/main-actions.js

```diff
@@ -34,7 +34,16 @@ export function createTab(location: ?string = undefined,
     const id = uuid.v4();
 
     // Start loading a tab while asking the User Agent to start its browsing session.
-    dispatch({ type: types.CREATE_TAB, id, ancestorId, location, options, instrument: true });
+    // Dispatching a list ensures only one render and prevents racing to set focus across user
```

Comment:
Here's my understanding:

Given our current configuration, using the thunk middleware we have... If two plain object actions are dispatched in a row, then no other action should be able to get in between them. A plain object action gets routed through our thunk middleware without dispatching anything else. So it get passed into Redux's dispatch, which [passes the state and action](https://github.com/reactjs/redux/blob/master/src/createStore.js#L170) to the root reducer synchronously and doesn't allow any actions to be dispatched in the reducers.

The one caveat is that if we introduce new custom middleware, it could theoretically do wild and unforseen things.